src/main/java/com/minecolonies/coremod/client/gui/WindowHireWorker.java

```diff
@@ -192,12 +192,33 @@ private void doneClicked(@NotNull final Button button)
         final int id = citizens.get(row).getId();
         @NotNull final ICitizenDataView citizen = citizens.get(row);
 
+        if (citizen.getWorkBuilding() != null) {
```

Comment:
even if you don't take the rest, someone should probably consider doing something like this for hiring from the library.